### PR TITLE
Add Loader Delay to Group Search

### PR DESCRIPTION
**Pull request by PCOffline**

### CODE REVIEW

File: mock-api/app.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 import cors from 'cors';
 import apiMocker from 'connect-api-mocker';
 
 const port = 4000;
 const app = express();
 
 app.use(cors());
-app.use(bodyParser({ limit: '50mb' }));
+app.use(express.urlencoded({ limit: '50mb', extended: true }));
+app.use(express.json({ limit: '50mb' }));
```
Comment (carmeltsadok) [inline mock-api/app.js:10]: Why?

File: src/components/search-group/scrollable-groups-result/ScrollableGroupsResult.jsx [hunk 2]
```diff
@@ -71,29 +76,36 @@ const ScrollableGroupsResult = ({
...
+
     return (
       <Fade in>
         <div className={classes.wrap}>
-          {currentUserGroups && currentUserGroups.length > 0 && (currentUserGroupsList)}
-          {privateGroups && privateGroups.length > 0 && (privateGroupsList)}
-          {publicGroups && publicGroups.length > 0 && (publicGroupsList)}
+          {currentUserGroups?.length && currentUserGroupsList}
+          {privateGroups?.length && privateGroupsList}
+          {publicGroups?.length && publicGroupsList}
```
Comment (carmeltsadok) [inline src/components/search-group/scrollable-groups-result/ScrollableGroupsResult.jsx:92]:
```suggestion
          {currentUserGroups?.length > 0 && currentUserGroupsList}
          {privateGroups?.length > 0 && privateGroupsList}
          {publicGroups?.length > 0 && publicGroupsList}
```

If you search only `מבצע סודי` for example, you will get `0` on public groups (and it shows it)

File: src/components/search-group/scrollable-groups-result/ScrollableGroupsResult.jsx [hunk 3]
```diff
@@ -20,6 +21,10 @@ const ScrollableGroupsResult = ({
 }) => {
   const classes = useStyles();
   const { t } = useTranslation();
+  const [showLoader, setShowLoader] = useState(false);
+  let timeout;
+
+  useEffect(() => () => clearTimeout(timeout));
```
Comment (carmeltsadok) [inline src/components/search-group/scrollable-groups-result/ScrollableGroupsResult.jsx:27]: When does it happen?

File: src/components/search-group/scrollable-groups-result/ScrollableGroupsResult.jsx [hunk 4]
```diff
@@ -71,29 +76,36 @@ const ScrollableGroupsResult = ({
...
+          {publicGroups?.length && publicGroupsList}
         </div>
       </Fade>
     );
   };
 
+  const renderContent = () => {
+    timeout = setTimeout(() => {
+      setShowLoader(isLoading);
+    }, 1000);
```
Comment (carmeltsadok) [inline src/components/search-group/scrollable-groups-result/ScrollableGroupsResult.jsx:101]: I think it should be inside an `useEffect` method.

**Review by carmeltsadok — CHANGES_REQUESTED** [review]
Great job!!

**Comment by PCOffline** [comment]
FYI: In my opinion the 1 second delay causes certain confusion and an illusion of lag in the app, I suggest decreasing the delay.

**Review by carmeltsadok — APPROVED** [review]

**Comment by carmeltsadok** [comment]
Great job. I would like you not to resolve messages. I prefer liking a message because I can't remember what I have said :)

**Pull request merged**